src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -102,17 +107,22 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
                         contextId, "company_delta");
                 cpDocument.setUpdated(updated);
             }
-            companyProfileRepository.save(cpDocument);
-        } catch (IllegalArgumentException illegalArgumentEx) {
-            throw new BadRequestException(illegalArgumentEx.getMessage());
+            ApiResponse<Void> response = insolvencyApiService.invokeChsKafkaApi(
+                    contextId, companyNumber);
+
+            if (response.getStatusCode() == HttpStatus.OK.value()) {
+                companyProfileRepository.save(cpDocument);
+            } else {
+                String errorMessage = String.format(
+                        "Error occurred while calling /resource-changed with "
+                        + "contextId %s and company number %s", contextId, companyNumber);
+                logger.error(errorMessage);
+                throw new RuntimeException(errorMessage);
+            }
         } catch (DataAccessException dbException) {
             throw new ServiceUnavailableException(dbException.getMessage());
         }
         logger.trace(String.format("Company profile is updated in MongoDB with contextId %s "
                 + "and company number %s", contextId, companyNumber));
-
-        insolvencyApiService.invokeChsKafkaApi(contextId, companyNumber);
-        logger.info(String.format("Chs-kafka-api CHANGED invoked successfully for "
-                + "contextId %s and company number %s", contextId, companyNumber));
```

Comment:
This log is specifically added as part of a logging requirement - please keep and move immediately after the successful `invokeChsKafkaApi` call.